src/client/components/pages/RoomAdmin/__tests__/RoomAdminTable.test.tsx

```diff
@@ -21,6 +24,186 @@ import {
+      const bodyRow = rows.filter((row) => (
+        within(row).queryAllByRole('columnheader').length === 0
+      ));
+      bodyRow.forEach((row) => {
+        strictEqual(
+          within(row).queryByText('Allston') !== null,
+          true,
+          'Allston is not in the table as expected'
+        );
+      });
+    });
+    it('filters table data when cambridge is selected', async function () {
+      await wait(() => renderResult.getAllByRole('row').length > 1);
+      const [campusDropDown] = renderResult.getAllByLabelText('The table will be filtered by selected campus name');
+      fireEvent.change(campusDropDown, { target: { value: 'Cambridge' } });
+      await wait(() => renderResult.getAllByRole('row').length > 1);
+      const rows = renderResult.getAllByRole('row');
+      const bodyRow = rows.filter((row) => (
+        within(row).queryAllByRole('columnheader').length === 0
+      ));
+      bodyRow.forEach((row) => {
+        strictEqual(
+          within(row).queryByText('Cambridge') !== null,
+          true,
+          'Cambridge is not in the table as expected'
+        );
+      });
+    });
+  });
+});
+describe('Building Dropdown', function () {
+  let getStub: SinonStub;
+  let renderResult: RenderResult;
+  beforeEach(function () {
+    getStub = stub(LocationAPI, 'getAdminRooms');
+    getStub.resolves(adminRoomsResponse);
+    renderResult = render(
+      <RoomAdmin />
+    );
+  });
+  it('defaults to all', async function () {
+    await wait(() => renderResult.getAllByRole('row').length > 1);
+    const buildingDropDown = renderResult.getByLabelText('The table will be filtered by selected building name', { exact: true }) as HTMLSelectElement;
+    strictEqual(buildingDropDown.value, 'All');
+  });
+  it('lists all available building options', async function () {
+    await wait(() => renderResult.getAllByRole('row').length > 1);
+    const buildingDropDown = renderResult.getByLabelText('The table will be filtered by selected building name');
+    const dropdownOptions = within(buildingDropDown)
+      .getAllByRole('option') as HTMLOptionElement[];
+    const buildingDropdownLabels = dropdownOptions
+      .map(({ textContent }) => textContent);
+    const buildingList = adminRoomsResponse.map((el) => el.building.name);
+    // slicing 'All' value as it's manually added default value not generated from adminRoomsResponse
+    strictEqual(
+      buildingDropdownLabels.slice(1).length === buildingList.length, true,
+      'building dropdown menu is not listing all available buildings'
+    );
+  });
+  context('when building dropdown value change', function () {
+    it('filters table data for selected value', async function () {
+      await wait(() => renderResult.getAllByRole('row').length > 1);
+      const [buildingDropDown] = renderResult.getAllByLabelText('The table will be filtered by selected building name');
+      fireEvent.change(buildingDropDown, { target: { value: 'Bauer Laboratory' } });
```

Comment:
I think it would be helpful to future proof the code by replacing "Bauer Laboratory" with a reference to a room from `adminRoomsResponse` (e.g. `adminRoomsResponse[0].building.name`). This way, if the test data is changed in the future such that `Bauer Laboratory` is removed, the test will still work, because the building you are changing it to references what we are stubbing `getAdminRooms` to resolve to on line 32: 
`getStub.resolves(adminRoomsResponse);`